static/js/narrow_state.js

```diff
@@ -340,6 +347,22 @@ export function excludes_muted_topics() {
     );
 }
 
+export function excludes_muted_users() {
+    // Indicates whether message lists should be filtered
+    // on the basis of user muting.
+    // Messages from muted users are hidden completely only when
+    // they are 1:1 PMs.
+
+    // Can this list contain 1:1 PMs?
+    return (
+        narrowed_to_search() ||
+        narrowed_to_starred() ||
+        narrowed_to_mentioned() ||
+        narrowed_to_pms() ||
+        !active()
+    );
+}
```

Comment:
It might be better to move this function to the class in `filter.js`, which has a really nice test suite where it'd be convenient to verify a new function like this.  And I think that's going to be what you want to call anyway.

Reply:
(I think with that structure, we can avoid passing a new parameter when creating MessageList objects, and just check the attribute of the associated `filter` object?  Not sure.)

Reply:
Yes, I think so. We could initialize the `MessageListData` objects with just the related `Filter` object in that case.
There's a bunch or related functions in `narrow_state.js`, `narrowed_to_pms()` for example. Should all of those be moved to the `Filter` class?

Reply:
(I have added tests for this function in the current file though)

Reply:
I don't think I understand this function; do we want something more like `return !narrowed_to_personal_pms` (pseudocode)?

We may also want a different name -- e.g. `shows_muted_user_messages` with the opposite meaning? Not sure, but generally when I see a function that lists 5 different cases like this, and don't have a clear explanation for why those are the 5 cases, I worry this is wrong.

(One detail you may not know is that you can generate a wide range of other possible narrows by just typing different combinations of operations into the search box; and `narrowed_to_search` specifically checks for a search keyword only)